File: Examples/db/qr/delphi/JpegData.pas
unit JpegData;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Db, DBTables;

type
  TDataModule1 = class(TDataModule)
    Table1: TTable;
    Table1Subject: TStringField;
    Table1Picture: TBlobField;
    procedure DataModule1Create(Sender: TObject);
    procedure DataModule1Destroy(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  DataModule1: TDataModule1;

implementation

{$R *.DFM}

procedure TDataModule1.DataModule1Create(Sender: TObject);
begin
  Table1.TableName := ExtractFilePath(Application.Exename)+Table1.TableName;
  if not FileExists(Table1.TableName) then
    Table1.CreateTable;
  Table1.Open

end;

procedure TDataModule1.DataModule1Destroy(Sender: TObject);
begin
     if Table1.State in [dsEdit, dsInsert] then Table1.Post
end;

end.

File: Examples/db/qr/cbuilder/jpegdata.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "jpegdata.h"
//---------------------------------------------------------------------------
#pragma resource "*.dfm"
TDataModule1 *DataModule1;
//---------------------------------------------------------------------------
__fastcall TDataModule1::TDataModule1(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TDataModule1::DataModule1Create(TObject *Sender)
{
  Table1->TableName = ExtractShortPathName(ExtractFilePath(Application->ExeName)+Table1->TableName);
  if (!FileExists(Table1->TableName))
    Table1->CreateTable();
  Table1->Open();
}
//---------------------------------------------------------------------------
void __fastcall TDataModule1::DataModule1Destroy(TObject *Sender)
{
     if (Table1->State == dsEdit || Table1->State == dsInsert)
       Table1->Post();
}
//---------------------------------------------------------------------------